File: frontend/src/types/report.ts
export type ReportTargetType = 'post' | 'comment' | 'user';
export type ReportStatus = 'pending' | 'reviewing' | 'resolved' | 'rejected';
export type ReportReason = 'spam' | 'harassment' | 'inappropriate' | 'copyright' | 'other';
export type ActionTaken = 'no_action' | 'content_blinded' | 'user_warned' | 'user_suspended';

export interface ReportCreateRequest {
  target_type: ReportTargetType;
  target_id: number;
  reason: ReportReason;
  description?: string;
}

export interface Report {
  id: number;
  reporter_id: number;
  reporter_name: string;
  target_type: ReportTargetType;
  target_id: number;
  reason: ReportReason;
  description: string | null;
  status: ReportStatus;
  review_started_by_id: number | null;
  review_started_by_name: string | null;
  review_started_at: string | null;
  resolved_by_id: number | null;
  resolved_by_name: string | null;
  resolution_note: string | null;
  resolved_at: string | null;
  action_taken: ActionTaken | null;
  created_at: string;
}

export interface ReportListResponse {
  items: Report[];
  total: number;
  pending_count: number;
}

export interface ReportStats {
  total: number;
  pending: number;
  reviewing: number;
  resolved: number;
  rejected: number;
}

export interface ReportResolveRequest {
  status: 'resolved' | 'rejected';
  resolution_note?: string;
  action_taken?: ActionTaken;
}

export interface ReportFilters {
  status?: ReportStatus;
  target_type?: ReportTargetType;
  reason?: ReportReason;
  skip?: number;
  limit?: number;
}

File: backend/app/schemas/report.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional
from app.models.report import ReportTargetType, ReportReason, ReportStatus


class ReportCreate(BaseModel):
    """Schema for creating a new report"""

    target_type: ReportTargetType
    target_id: int
    reason: ReportReason
    description: Optional[str] = Field(None, max_length=1000)


class ReportOut(BaseModel):
    """Schema for report response"""

    id: int
    reporter_id: int
    reporter_name: Optional[str] = None
    target_type: ReportTargetType
    target_id: int
    reason: ReportReason
    description: Optional[str]
    status: ReportStatus
    review_started_by_id: Optional[int]
    review_started_by_name: Optional[str] = None
    review_started_at: Optional[datetime]
    resolved_by_id: Optional[int]
    resolved_by_name: Optional[str] = None
    resolution_note: Optional[str]
    resolved_at: Optional[datetime]
    action_taken: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True


class ReportResolve(BaseModel):
    """Schema for resolving a report"""

    status: ReportStatus
    resolution_note: Optional[str] = Field(None, max_length=1000)
    action_taken: Optional[str] = Field(None, max_length=100)
    # Actions: "content_blinded", "user_warned", "user_suspended", "no_action"


class ReportListOut(BaseModel):
    """Schema for paginated report list"""

    items: list[ReportOut]
    total: int
    pending_count: int


class ReportStats(BaseModel):
    """Schema for report statistics"""

    total: int
    pending: int
    reviewing: int
    resolved: int
    rejected: int
